src/storage.ts

```diff
@@ -20,3 +22,30 @@ export class PinataStorage implements Storage {
         })).IpfsHash;
     }
 }
+
+export class S3Storage implements Storage {
+    private readonly bucket: string;
+    private readonly s3: S3;
+
+    constructor(accessKeyId: string, secretAccessKey: string, bucket: string) {
+        this.bucket = bucket;
+        this.s3 = new S3({
+            credentials: {
```

Comment:
This object can be written as just
```ts
{
                accessKeyId,
                secretAccessKey,
}
```
(no need to put `!` and do type assertions)